cmds/mktemp/mktemp.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Mktemp makes a temporary file (or directory)
+//
+// Synopsis:
+//     mktemp [-d] [-p]
+//
+// Options:
+//     -d: make a temp directory instead of file
+//     -p: change prefix used for template
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+var (
+	directory = flag.Bool("d", false, "Create a directory instead of file")
+	prefix    = flag.String("p", "", "Add prefix to end of temp creation")
+)
+
+func init() {
+	// Usage Definition
+	defUsage := flag.Usage
+	flag.Usage = func() {
+		os.Args[0] = "mktemp [-dp]"
+		defUsage()
+	}
+}
+
+func main() {
+	flag.Parse()
+	if len(flag.Args()) > 0 {
```

Comment:
most uses of mktemp I find expect to have an argument for a template, e.g.
/etc/libpaper.d/texlive-base:    tempfile=$(mktemp -p /tmp mktexlsr.XXXXXXXX)
/etc/libpaper.d/texlive-base:    tempfile=$(mktemp -p /tmp fmtutil.XXXXXXXX)
You could look at what I did in #794, namely, take an arg, split it on 'X', and use what remains as the template. 

Otherwise this mktemp can't be used in most scripts.

Reply:
@rminnich Sorry for stepping on your toes here. Really like what you had going on on your PR. This is now updated with mostly your changes, some tests, etc